src/shims/unix/fs.rs

```diff
@@ -272,8 +272,6 @@ trait EvalContextExtPrivate<'tcx>: crate::MiriInterpCxExt<'tcx> {
     ) -> InterpResult<'tcx, i32> {
         let this = self.eval_context_mut();
 
-        let mode: u16 = metadata.mode.to_u16()?;
```

Comment:
This should be `metadata.mode.to_uint(this.libc_ty_layout("mode_t").size)`. Then the `mode.into()` below should become `mode.try_into().unwrap()`.